api/server.go

```diff
@@ -104,6 +105,13 @@ func Serve(ctx context.Context, serveAddr string, db idb.IndexerDb, dataError fu
 
 	e.Use(middlewares.MakeLogger(log))
 	e.Use(middleware.CORS())
+	e.Use(middleware.GzipWithConfig(middleware.GzipConfig{
+		// we currently support compressed result only for GET /v2/blocks/ API
+		Skipper: func(c echo.Context) bool {
+			return !strings.Contains(c.Path(), "/v2/blocks/") || c.QueryParam("compress") != "true"
+		},
```

Comment:
I limited the ability to use gzip just for the `/v2/blocks` endpoint because I think we need to add tests for all endpoints. In order to avoid that and just address the problem we have, I think it is a good start. If you still think we should support all endpoints I can do that, but adding tests for all endpoints might take a while.

As for your second point, do you suggest we use gzip every time the client support `gzip` encoding? even if it didn't request it? My approach was only to add the support and not replace it (don't want to break any 3rd party code)

Reply:
That's fair, let's limit it to the blocks endpoint for now then.

Existing code probably isn't supplying the `Accept-Encoding: gzip` header. Without specifying the header explicitly, the middleware will won't compress anything, so I expect almost all clients to have the same behavior by default.